src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -8,393 +8,109 @@
-	{
-		if(feats.randomcoeff_multiplicative==NULL)
-		{
-			throw ShogunException(
-							"void CRandomFourierGaussPreproc::copy(...): feats.randomcoeff_multiplicative==NULL && cur_dim_feature_space>0 &&(cur_dim_input_space>0)  \n");
-		}
-
-		randomcoeff_multiplicative=SG_MALLOC(float64_t, cur_dim_feature_space*cur_dim_input_space);
-		std::copy(feats.randomcoeff_multiplicative,feats.randomcoeff_multiplicative+cur_dim_feature_space*cur_dim_input_space,randomcoeff_multiplicative);
-	}
-	else
-	{
-		randomcoeff_multiplicative = NULL;
-	}
-
-}
-
-RandomFourierGaussPreproc::RandomFourierGaussPreproc() :
-	RandomMixin<DensePreprocessor<float64_t>> () {
-	dim_feature_space = 1000;
-	dim_input_space = 0;
-	cur_dim_input_space = 0;
-	cur_dim_feature_space=0;
-
-	randomcoeff_multiplicative=NULL;
-	randomcoeff_additive=NULL;
-
-	kernelwidth=1;
-	cur_kernelwidth=kernelwidth;
-
-	SG_ADD(&dim_input_space, "dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&cur_dim_input_space, "cur_dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&dim_feature_space, "dim_feature_space",
-			"Dimensionality of the feature space.");
-	SG_ADD(&cur_dim_feature_space, "cur_dim_feature_space",
-			"Dimensionality of the feature space.");
-
-	SG_ADD(&kernelwidth, "kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-	SG_ADD(&cur_kernelwidth, "cur_kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-
-	watch_param(
-		"randomcoeff_additive", &randomcoeff_additive,
-		&cur_dim_feature_space);
-
-	watch_param(
-		"randomcoeff_multiplicative", &randomcoeff_multiplicative,
-		&cur_dim_feature_space, &cur_dim_input_space);
-
-}
-
-RandomFourierGaussPreproc::RandomFourierGaussPreproc(
-		const RandomFourierGaussPreproc & feats) :
-	RandomMixin<DensePreprocessor<float64_t>> () {
-
-	randomcoeff_multiplicative=NULL;
-	randomcoeff_additive=NULL;
-
-	SG_ADD(&dim_input_space, "dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&cur_dim_input_space, "cur_dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&dim_feature_space, "dim_feature_space",
-			"Dimensionality of the feature space.");
-	SG_ADD(&cur_dim_feature_space, "cur_dim_feature_space",
-			"Dimensionality of the feature space.");
-
-	SG_ADD(&kernelwidth, "kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-	SG_ADD(&cur_kernelwidth, "cur_kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-
-	watch_param(
-		"randomcoeff_additive", &randomcoeff_additive,
-		&cur_dim_feature_space);
-
-	watch_param(
-		"randomcoeff_multiplicative", &randomcoeff_multiplicative,
-		&cur_dim_feature_space, &cur_dim_input_space);
-
-	copy(feats);
-}
-
-RandomFourierGaussPreproc::~RandomFourierGaussPreproc() {
-
-	SG_FREE(randomcoeff_multiplicative);
-	SG_FREE(randomcoeff_additive);
+RandomFourierGaussPreproc::RandomFourierGaussPreproc()
+: m_kernelwidth(1), m_dim_output(100)
+{
+	SG_ADD(
+	    &m_dim_output, "m_dim_output",
```

Comment:
could you remove the "m_" prefix? We don't usually add this (see the code you deleted)